src/uct/tcp/tcp_ep.c

```diff
@@ -478,7 +478,9 @@ ucs_status_t uct_tcp_ep_create(const uct_ep_params_t *params,
 
 done:
     iface->ep_count++;
-    ucs_assert(iface->ep_count != UCT_TCP_EP_ID_MAX);
+    if (iface->ep_count == UCT_TCP_EP_ID_MAX) {
```

Comment:
what if ep 0 is always alive, and there is a loop of 64k create/destroy in addition?
there will be a collision of ids

Reply:
> what if ep 0 is always alive, and there is a loop of 64k create/destroy in addition?
> there will be a collision of ids

yes, this is a problem.
let's use 64-bit value, so number of the connection will be limited by `2^64 - 1`